clarify no subfolders and md only for collections /docs folder (#63808)
ansible/ansible: docs/docsite/rst/dev_guide/developing_collections.rst

docs directory
---------------

> Keep general documentation for the collection here. Plugins and modules still keep their specific documentation embedded as Python docstrings. Use the ``docs`` folder to describe how to use the roles and plugins the collection provides, role requirements, and so on. Markdown is the standard format for documentation files in the ``docs`` folder. No subfolders will be supported on Galaxy.

Reviewer: I'd keep this present-tense. Who knows what we will do in future?
```suggestion
Keep general documentation for the collection here. Plugins and modules still keep their specific documentation embedded as Python docstrings. Use the ``docs`` folder to describe how to use the roles and plugins the collection provides, role requirements, and so on. Markdown is the standard format for documentation files in the ``docs`` folder. Galaxy does not support or process subfolders within the ``docs`` folder.
```
